test/explorer/series_test.exs

```diff
@@ -5866,4 +5866,20 @@ defmodule Explorer.SeriesTest do
       end
     end
   end
+
+  describe "json_decode/2" do
+    test "extracts struct from json" do
+      s = Series.from_list(["{\"n\": 1}"])
+      sj = Series.json_decode(s, {:struct, %{"n" => {:s, 64}}})
+      assert sj.dtype == {:struct, %{"n" => {:s, 64}}}
+      assert Series.to_list(sj) == [%{"n" => 1}]
+    end
+
+    test "extracts struct from json with dtype" do
+      s = Series.from_list(["{\"n\": 1}"])
+      sj = Series.json_decode(s, {:struct, %{"n" => {:f, 64}}})
+      assert sj.dtype == {:struct, %{"n" => {:f, 64}}}
+      assert Series.to_list(sj) == [%{"n" => 1.0}]
+    end
```

Comment:
Let's add a test for the dtype not matching. Let's also add the same tests but inside a DF.mutate (i.e. lazy).

Reply:
I just realised json_decode decodes to even primitives not just :struct and :list. It coerces to null for incompatible dtype. Error occurs when the string is not valid json.

```python
df = pl.DataFrame([{"a": "[\"a\"]"}])
df = df.with_columns(aj=pl.col("a").str.json_decode(dtype=pl.Int64,infer_schema_length=10), ajl=pl.col("a").str.json_decode(dtype=pl.List(pl.String),infer_schema_length=10))
shape: (1, 3)
┌───────┬──────┬───────────┐
│ a     ┆ aj   ┆ ajl       │
│ ---   ┆ ---  ┆ ---       │
│ str   ┆ i64  ┆ list[str] │
╞═══════╪══════╪═══════════╡
│ ["a"] ┆ null ┆ ["a"]     │
└───────┴──────┴───────────┘
```

Reply:
Excellent!